File: python/9_16_2022/binary_search.py
def do_binary_search(arr, val):
    low = 0
    high = len(arr)
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] > val:
            high = mid - 1
        elif arr[mid] < val:
            low = mid + 1
        else:
            return mid
    return -1

File: java/d9232022/src/main/java/com/example/BinarySearch.java
package com.example;

public class BinarySearch {
    public int doBinarySearch(int arr[], int val){
        int low = 0;
        int high = arr.length;
        while (low <= high){
            int mid = (low+high)/2;
            if (arr[mid] > val){
                high = mid - 1;
            }
            else if(arr[mid] < val){
                low = mid + 1;
            }
            else{
                return mid;
            }
        }
        return -1;
    }
}
